src/modules/blockchainApplication/connection/components/ConnectionSummary/index.test.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mountWithRouter } from 'src/utils/testHelpers';
+import usePairings from '@libs/wcm/hooks/usePairings';
+import useSession from '@libs/wcm/hooks/useSession';
+import { EVENTS } from '@libs/wcm/constants/lifeCycle';
+import ConnectionContext from '@libs/wcm/context/connectionContext';
+import ConnectionSummary from './index';
+
+jest.mock('@libs/wcm/hooks/usePairings');
+jest.mock('@libs/wcm/hooks/useSession');
+jest.mock('@account/hooks', () => ({
+  useAccounts: jest.fn().mockImplementation(() => ({
+    accounts: [{ metadata: { address: 'lskdxc4ta5j43jp9ro3f8zqbxta9fn6jwzjucw7yt' } }],
+  })),
+}));
+
+jest.mock('@walletconnect/utils', () => ({
+  getSdkError: jest.fn(str => str),
+}));
+
+jest.spyOn(React, 'useContext').mockImplementation(() => ({
+  events: [{ name: EVENTS.SESSION_PROPOSAL, meta: { id: '1' } }],
+}));
+
+jest.mock('@libs/wcm/utils/connectionCreator', () => ({
+  createSignClient: jest.fn(() => Promise.resolve()),
+  client: {
+    pair: jest.fn(),
+  },
+}));
+
+const setup = (context) => {
+  const Component = () => (
+    <ConnectionContext.Provider value={context}>
+      <ConnectionSummary />
+    </ConnectionContext.Provider>
+  );
+
+  return mountWithRouter(Component, {}, {});
+};
+
+describe('ConnectionSummary', () => {
+  const approve = jest.fn();
+  const reject = jest.fn();
+  usePairings.mockReturnValue({ setUri: jest.fn() });
+  useSession.mockReturnValue({ approve, reject });
+
+  const proposal = {
+    params: {
+      proposer: {
+        metadata: {
+          name: 'Proposer name',
+          url: 'http://example.com',
+          icons: ['http://example.com/icon.png'],
+        },
+      },
+      requiredNamespaces: {
+        lisk: {
+          chains: ['lsk:1'],
+          events: ['sign_transaction'],
+          methods: ['receive_token'],
+        },
+      },
+      pairingTopic: '0x123',
+    },
+  };
+  const context = {
+    events: [{ name: EVENTS.SESSION_PROPOSAL, meta: proposal }],
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('Display the connecting app information and connection summary', () => {
+    const wrapper = setup(context);
+    expect(wrapper.find('.chain-name-text').text()).toEqual(proposal.params.proposer.metadata.name);
+    expect(wrapper.find('img').at(0).prop('src')).toEqual(proposal.params.proposer.metadata.icons[0]);
+    // expect(wrapper.find('.accounts').text()).toEqual(proposal.params.accounts[0].address);
```

Comment:
I think this should be removed if its not in `TODO:` and if it is then a to `TODO` description can be provided

Reply:
Done